src/nfa/nfa.cc

```diff
@@ -626,7 +613,7 @@ Nfa Mata::Nfa::remove_epsilon(const Nfa& aut, Symbol epsilon)
     }
 
     bool changed = true;
-    const TransSymbolStates epsilon_symbol_to_find{ epsilon };
+    const Move epsilon_symbol_to_find{epsilon };
```

Comment:
Haven't noticed this before, but, what the fuck is this variable? It's Move, but its epsilon symbol? Like whaaaat?

Reply:
Ah, yeah. A beautiful example of how the `OrdVector` (behaving as `std::set` or `std::vector`) can use `std::set` operations, but only on the data type it contains. Therefore, the key to method `find` has to be a Move with prefilled epsilon symbol in order for the method `find` to accept it and actually find the move with epsilon. You cannot just pass a symbol to this method and expect it to be able to find a Move with the symbol as the data types differ.

We can either reimplement this, seeing as looking for moves with a certain symbol is a commonly used operation, or we must create a dummy Move structure each time, as you can see here.

Reply:
I think this is a acceptable price for not introducing some kind of crazy polymorphism. It is not nice, but I hope that everyone can understand why it is coded this way. Maybe we can add a comment here?

Reply:
I'm rather criticizing the naming :-), I'm fine with the code. But it's kindof like `std::vector<int> single_number`. The Move consists of transitions, but the variable is called symbol :D

Reply:
Maybe call it `epsilon_symbol_wrapper`?

Reply:
that sounds good to me.

Reply:
Actually, the variable has only one usage few line bellow the definition. What about replacing it there simply by `Move{epsilon}`?

Reply:
The comment could be useful, but then again, we would need to add this comment each time we hack our way around this anywhere in the code. Therefore, I would refrain from adding a comment and instead explicitly warn the user that this is a wrapper around some specific behaviour, as suggested by @tfiedor, instead.

Reply:
Yes, that should be OK as well.

Reply:
Fine by me.